### PR TITLE
Prometheus support

### PR DESCRIPTION
**Pull request by fbalak**
- Add Prometheus API
- Add simple `test_alerting_works` test
- Add `login` and `get_user_token` functions to `OCP` class
- Add `username` and `password_location` to `default_config.yaml`
- Generate CA certificate for API usage

### CODE REVIEW

**Comment by fbalak** [comment]
Test logs:
```
Collecting Cluster versions
======================================================= test session starts ========================================================
platform linux -- Python 3.6.6, pytest-4.6.3, py-1.8.0, pluggy-0.12.0 -- /home/ocsqe/data/fbalak/ocs-ci/venv3/bin/python3
cachedir: .pytest_cache
metadata: {'Cluster Version': '4.1.4', 'Ceph Version': '14.2.1 (d555a9489eb35f84f2e1ef49b77e19da9d113972) nautilus (stable)', 'Rook Version': 'v1.0.0-196.g1a7766b', 'csi-provisioner': 'v1.2.0', 'cephfsplugin': None, 'rbdplugin': None}
rootdir: /home/ocsqe/data/fbalak/ocs-ci, inifile: pytest.ini
plugins: reportportal-1.0.5, metadata-1.8.0, html-1.21.1, logger-0.5.0
collected 1 item                                                                                                                   

tests/test_alerting_works.py::test_alerting_works 
---------------------------------------------------------- live log setup ----------------------------------------------------------
13:32:28 - MainThread - tests.conftest - INFO - All logs located at /tmp/logs_1562758337
13:32:28 - MainThread - tests.conftest - INFO - Running OCS basic installation
13:32:28 - MainThread - ocs_ci.ocs.openshift_ops - INFO - Testing access to cluster with /home/ocsqe/data/ocs-instances/current/auth/kubeconfig
13:32:28 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc cluster-info
13:32:29 - MainThread - ocs_ci.ocs.openshift_ops - INFO - Access to cluster is OK!
13:32:29 - MainThread - tests.conftest - INFO - The installation is skipped because the cluster is running
---------------------------------------------------------- live log call -----------------------------------------------------------
13:32:29 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc login -u kubeadmin -p IXCxX-cLGrB-9LMZB-NvqGY
13:32:32 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring --kubeconfig /home/ocsqe/data/ocs-instances/current/auth/kubeconfig whoami --show-token
13:32:32 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring --kubeconfig /home/ocsqe/data/ocs-instances/current/auth/kubeconfig get Route prometheus-k8s -o yaml
13:32:33 - MainThread - /home/ocsqe/data/fbalak/ocs-ci/ocs_ci/utility/prometheus.py - INFO - Generated CA certification file: /tmp/tmpcg1_yx_f
13:32:33 - MainThread - /home/ocsqe/data/fbalak/ocs-ci/ocs_ci/utility/prometheus.py - INFO - GET https://█████████████████████████.com/api/v1/alerts
13:32:33 - MainThread - /home/ocsqe/data/fbalak/ocs-ci/ocs_ci/utility/prometheus.py - INFO - headers={'Authorization': 'Bearer ████████████████████████'}
13:32:33 - MainThread - /home/ocsqe/data/fbalak/ocs-ci/ocs_ci/utility/prometheus.py - INFO - verify=/tmp/tmpcg1_yx_f
13:32:33 - MainThread - /home/ocsqe/data/fbalak/ocs-ci/ocs_ci/utility/prometheus.py - INFO - params={'silenced': False, 'inhibited': False}
13:32:34 - MainThread - tests.test_alerting_works - INFO - Prometheus Alerts: [{'labels': {'alertname': 'Watchdog', 'severity': 'none'}, 'annotations': {'message': 'This is an alert meant to ensure that the entire alerting pipeline is functional.\nThis alert is always firing, therefore it should always be firing in Alertmanager\nand always fire against a receiver. There are integrations with various notification\nmechanisms that send a notification when this alert is not firing. For example the\n"DeadMansSnitch" integration in PagerDuty.\n'}, 'state': 'firing', 'activeAt': '2019-07-10T07:20:30.163677339Z', 'value': 1}]
PASSED

===================================================== 1 passed in 5.77 seconds =====================================================
```

File: tests/test_alerting_works.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import logging
```
Comment (vasukulkarni) [inline tests/test_alerting_works.py]: The file should be under tests/UI ?

File: ocs_ci/utility/prometheus.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        cert_file.write(
+            base64.b64decode(
+                kubeconfig['clusters'][0]['cluster']['certificate-authority-data']
+            )
+        )
+        cert_file.close()
+        self._cacert = cert_file.name
+        logger.info(f"Generated CA certification file: {self._cacert}")
+
+    def alerts(self, silenced=False, inhibited=False):
```
Comment (vasukulkarni) [inline ocs_ci/utility/prometheus.py]: the function can be called Get and pattern could be its argument so that you can reuse it for other Get's too

Comment (fbalak) [reply ocs_ci/utility/prometheus.py]: Done

File: tests/test_alerting_works.py (hunk 1 above)
Comment (fbalak) [reply tests/test_alerting_works.py]: Moved, although most of alerting tests will be probably under `tests/manage` directory.

**Comment by ebenahar** [comment]
@fbalak please re-execute tests/test_alerting_works.py::test_alerting_works for verification

**Comment by keesturam** [comment]
@Akarsha-rai fyi. please have a look at this PR.

**Comment by keesturam** [comment]
Looks good.

File: ocs_ci/utility/prometheus.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            )
+            if os.path.exists(filename):
+                with open(filename) as f:
+                    password = f.read()
+            else:
+                raise FileNotFoundError(
+                    errno.ENOENT,
+                    os.strerror(errno.ENOENT),
+                    filename
+                )
```
Comment (mbukatov) [inline ocs_ci/utility/prometheus.py]: You don't need to test existence of the file, if the only thing you are going to do is to open it and raise `FileNotFoundError` exception if the file doesn't exist manually. Is there any reason why you can't just open and read the file like shown below without the if - else - raise wrapper code?

```
                with open(filename) as f:
                    password = f.read()
```

Comment (mbukatov) [reply ocs_ci/utility/prometheus.py]: See:

```
$ ipython3 
Python 3.7.3 (default, May 11 2019, 00:38:04) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.2.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: with open("nenineni") as f: 
   ...:     print(f.read()) 
   ...:                  
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-1-1afc67e7ddb6> in <module>
----> 1 with open("nenineni") as f:
      2     print(f.read())
      3 

FileNotFoundError: [Errno 2] No such file or directory: 'nenineni'
```

Comment (fbalak) [reply ocs_ci/utility/prometheus.py]: Good point. Fixed.

**Pull request merged**